### DISCUSSION TITLE
Feature Request: Add Purchase Limit Per Player for QuickShop

### DISCUSSION THREAD

**Discussion by EnTIv** [QuickShop-Community/QuickShop-Hikari#1768] [Ideas]
Description:

I would like to request a feature that allows shop owners to set a purchase limit per player for a specific time period. This would help shop owners sell items at a lower price without worrying about players buying in bulk or taking advantage of temporary discounts.

Proposed Feature:

Add a setting for shop owners to limit how many items each player can purchase from a shop within a specific timeframe.
For example, a shop owner could set a limit where each player can only buy 5 diamonds every 30 minutes.
This limit should be customizable per shop, per player, and per time period (e.g., 5 items per player per 30 minutes).
Use Case:

Shop owners could run promotional events where items are sold below market value.
By setting a purchase limit, this prevents bulk purchases from a single player and ensures fair access to the promotion.
Helps maintain balance in player economies by limiting reselling or monopolizing limited offers.
Benefits:

More control for shop owners on how items are distributed.
Encourages fair play and reduces the chance of one player dominating the purchase of discounted goods.
Can help create more exciting and competitive events in the server's economy.
Thank you for considering this request! It would be a great addition to the QuickShop plugin and improve the overall user experience.

**Comment by creatorfromhell** [comment]
> Description:
> 
> I would like to request a feature that allows shop owners to set a purchase limit per player for a specific time period. This would help shop owners sell items at a lower price without worrying about players buying in bulk or taking advantage of temporary discounts.
> 
> Proposed Feature:
> 
> Add a setting for shop owners to limit how many items each player can purchase from a shop within a specific timeframe. For example, a shop owner could set a limit where each player can only buy 5 diamonds every 30 minutes. This limit should be customizable per shop, per player, and per time period (e.g., 5 items per player per 30 minutes). Use Case:
> 
> Shop owners could run promotional events where items are sold below market value. By setting a purchase limit, this prevents bulk purchases from a single player and ensures fair access to the promotion. Helps maintain balance in player economies by limiting reselling or monopolizing limited offers. Benefits:
> 
> More control for shop owners on how items are distributed. Encourages fair play and reduces the chance of one player dominating the purchase of discounted goods. Can help create more exciting and competitive events in the server's economy. Thank you for considering this request! It would be a great addition to the QuickShop plugin and improve the overall user experience.

This sounds like the limit addon we already have.
